pkg/controller/controller_binding.go

```diff
@@ -570,17 +617,26 @@ func (c *controller) reconcileServiceInstanceCredential(binding *v1alpha1.Servic
 			"The binding was deleted successfully",
 		)
 		c.clearServiceInstanceCredentialCurrentOperation(toUpdate)
-		// Clear the finalizer
-		finalizers.Delete(v1alpha1.FinalizerServiceCatalog)
-		toUpdate.Finalizers = finalizers.List()
+		if toUpdate.Status.OrphanMitigationInProgress {
+			s := fmt.Sprintf(`Orphan mitigation successful for ServiceInstanceCredential "%v/%v", setting condition to failed`, binding.Namespace, binding.Name)
+			c.setServiceInstanceCredentialCondition(toUpdate,
```

Comment:
I don't think we want to overwrite the reason and message for the Failed condition.